dexmaker/src/main/java/com/android/dx/MethodId.java

```diff
@@ -66,7 +66,15 @@ public boolean isConstructor() {
     }
 
     /**
-     * Returns the method's name. This is "&lt;init&gt;" if this is a constructor.
+     * Returns true if this method is the static initializer for its declaring class.
+     */
+    public boolean isStaticInitializer() {
+        return name.equals("<clinit");
```

Comment:
This is missing the closing `>`, which means the `ACC_CONSTRUCTOR` flag won't get added to the method.

I did some experimenting to see if there's a way we can modify your test to catch this, but it looks like on both Dalvik and ART (I tested api 19 and 22), Android will run the `clinit` method properly as long as it's marked `static`...it doesn't seem to matter if it's not marked as a `constructor`.

Since you can't get ahold of the static initializer via reflection, I don't see any way to test this case, so I guess we can just fix the typo and move on 😄